query-engine/connector-test-kit-rs/query-tests-setup/src/datamodel_rendering/mod.rs

```diff
@@ -89,8 +91,13 @@ fn process_template(template: String, renderer: Box<dyn DatamodelRenderer>) -> S
     })
 }
 
-fn render_preview_features(excluded_features: &[&str]) -> String {
-    let excluded_features: Vec<_> = excluded_features.iter().map(|f| format!(r#""{f}""#)).collect();
+fn render_preview_features(excluded_features: &[&str], version: &ConnectorVersion) -> String {
+    let mut excluded_features: Vec<_> = excluded_features.iter().map(|f| format!(r#""{f}""#)).collect();
+
+    // TODO: Remove this once we are able to have version speficic preview features.
+    if version.is_mysql() && !matches!(version, ConnectorVersion::MySql(Some(MySqlVersion::V8))) {
+        excluded_features.push(format!(r#""{}""#, PreviewFeature::RelationJoins));
+    }
```

Comment:
To avoid having to exclude hundreds of tests for MySQL 5.6 & 5.7, we simply do not render the `relationJoins` preview feature for now when rendering the datamodel of each tests. This ensure the `relationLoadStrategy: join` is never used.